File: Java/gesbrasilburger/src/main/java/sn/brasilburger/entity/Commande.java
package sn.brasilburger.entity;

import java.time.LocalDateTime;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Commande {
    private int id;
    private LocalDateTime dateCommande;
    private EtatCommande etat;
    private TypeCommande type;
    private Double montantTotal;
    private User client;
    private Zone zone;
    private Livreur livreur;
    private List<CommandeItem> items;
    private Paiement paiement;
}

File: CSharp/brasilBurger/Models/Commande.cs
namespace brasilBurger.Models
{
    public enum TypeCommande
    {
        SUR_PLACE,
        A_RECUPERER,
        LIVRAISON
    }
    public enum EtatCommande
    {
        EN_ATTENTE,
        EN_COURS,
        TERMINEE,
        ANNULEE
    }
    public class Commande
    {
        public int Id { get; set; }
        public DateTime DateCommande { get; set; }
        public EtatCommande Etat { get; set; }
        public TypeCommande Type { get; set; }
        public decimal MontantTotal { get; set; }
        public int ClientId { get; set; }
        public User Client { get; set; }
        public int? ZoneId { get; set; }
        public Zone Zone { get; set; }
        public int? LivreurId { get; set; }
        public Livreur Livreur { get; set; }
        public Paiement PaiementC { get; set; }
        public ICollection<CommandeItem> CommandeItems { get; set; } = new List<CommandeItem>();
    }
}